src/render.rs

```diff
@@ -141,11 +142,31 @@ impl<'a> Default for MarkdownRenderer<'a> {
 /// let text = "[Rust](https://rust-lang.org/) is an awesome *systems programming* language!";
 /// let rendered = markdown_to_html(text)?;
 /// ```
-pub fn markdown_to_html(text: &str) -> CargoResult<String> {
+fn markdown_to_html(text: &str) -> CargoResult<String> {
     let renderer = MarkdownRenderer::new();
     renderer.to_html(text)
 }
 
+static MARKDOWN_EXTENSIONS: [&'static str; 7] = [
+    ".md",
+    ".markdown",
+    ".mdown",
+    ".mdwn",
+    ".mkd",
+    ".mkdn",
+    ".mkdown",
+];
+
+pub fn readme_to_html(text: &str, filename: &str) -> CargoResult<String> {
+    for e in &MARKDOWN_EXTENSIONS {
+        if filename.to_lowercase().ends_with(e) {
+            return markdown_to_html(text);
+        }
+    }
+
+    Ok(encode_minimal(text).replace("\n", "<br>\n"))
```

Comment:
Why are we pulling `htmlescape` as a new dependency ? Isn't `ammonia` already doing a similar job ?

Reply:
`htmlescape` is replacing characters with their entities in order to basically turn text into HTML that represents/renders as the same text; `ammonia` is parsing HTML itself and then sanitizing that DOM. They're pretty different (and unfortunately, I looked and neither `ammonia` nor any of its dependencies, like, `html5ever`, can do the job of `htmlescape` simply.)

Reply:
Fine :)